backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_multibit.cu

```diff
@@ -624,14 +624,28 @@ void cuda_tbc_multi_bit_programmable_bootstrap_lwe_ciphertext_vector(
         lwe_dimension, polynomial_size, grouping_factor, base_log, level_count,
         num_samples, lut_count, lut_stride);
     break;
-  case 2048:
-    host_tbc_multi_bit_programmable_bootstrap<Torus, AmortizedDegree<2048>>(
-        static_cast<cudaStream_t>(stream), gpu_index, lwe_array_out,
-        lwe_output_indexes, lut_vector, lut_vector_indexes, lwe_array_in,
-        lwe_input_indexes, bootstrapping_key, pbs_buffer, glwe_dimension,
-        lwe_dimension, polynomial_size, grouping_factor, base_log, level_count,
-        num_samples, lut_count, lut_stride);
+  case 2048: {
+    int num_sms = 0;
+    check_cuda_error(cudaDeviceGetAttribute(
+        &num_sms, cudaDevAttrMultiProcessorCount, gpu_index));
+
+    if (4 * num_sms < num_samples * level_count * (glwe_dimension + 1))
```

Comment:
why do we need this if only in the 2048 case?

Reply:
We have no experimental results yet for other sizes to decide we need it or not, and for now only 2048 matters